Optik/Classes/Animation/AnimatableProperty.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  AnimatableProperty.swift
+//  Optik
+//
+//  Created by Htin Linn on 7/24/16.
+//  Copyright © 2016 Prolific Interactive. All rights reserved.
+//
+
+import UIKit
+
+/**
+ *  Defines an animatable property for the animation system.
+ */
+internal protocol AnimatableProperty {
+    
+    /// Underlying property type.
+    associatedtype PropertyType: VectorRepresentable
+    
+    /// Threshold value to use in determining if an animation is considered complete.
+    var threshold: CGFloat { get }
+    
+    /// Lens for reading and writing values.
+    var lens: Lens<UIView, PropertyType.InterpolatableType> { get }
+    
+}
+
+/**
+ *  View frame animatable property.
+ */
+internal struct ViewFrame: AnimatableProperty {
+    
+    typealias PropertyType = CGRect
+    
+    let threshold: CGFloat = 0.1
+    let lens: Lens<UIView, PropertyType.InterpolatableType> = Lens(
+        get: { $0.frame.vector },
+        set: { $1.frame = CGRect(vector: $0); return $1 }
+    )
+    
+}
+
+/**
+ *  View alpha animatable property.
+ */
+internal struct ViewAlpha: AnimatableProperty {
+    
+    typealias PropertyType = CGFloat
+    
+    let threshold: CGFloat = 0.01
```

Comment:
The threshold property is used to determine when the animatable values are considered close enough to stop animation. For example, if you are animating an `alpha` from 0 to 1, the animator will stop the animation when it sees that `alpha` is between 0.99 and 1.01. Obviously, we can use a smaller threshold here but then the animation will take longer and you can barely see the difference. 

The threshold value for `ViewFrame` is larger than `ViewAlpha` because `frame` values are usually larger than `alpha` values.